### PR TITLE
Input Search Dark Mode Friendly

### LINKED ISSUE #5734: Dark-mode friendly support for search input field
### Current Behavior
See the https://layer5.io/blog page and it's use of an input field while in dark mode.
See also the site home page and https://layer5.io/cloud-native-management/meshery/integrations for additional uses of a search input field.

Be sure to adhere to the Acceptance Tests below.

### Desired Behavior

Screenshot 2024-07-26 at 4 50 08 PM

### Acceptance Tests
1. All search input fields for follow the same dark-mode friendly behavior.
    1. [Integrations](https://layer5.io/cloud-native-management/meshery/integrations) page
    2. Home page
    3. Blog page

---
<h2>Contributor Resources and <a href="https://layer5.io/community/handbook">Handbook</a></h2>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- 📚 See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md).
-  🎨 Wireframes and [designs for Layer5 site](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.layer5.io).

Join the Layer5 Community by submitting your [community member form](https://layer5.io/newcomer).

### PR DESCRIPTION
**Pull request by THEBOSS0369**
This PR fixes #5734

**Description**

I have added some changes so that the input looks like more dark friendly. The changes for style has been done for **Intergration Page**, **Home Page** and **Blog Page**. The style is same as the style in **layer5/news** page. Below are the ss for the changes made.

main page
main page
blog

**Notes for Reviewers**
This PR is same as this one https://github.com/layer5io/layer5/pull/5881 . It is made because of DCO error not resolving.

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit ae8e3af6ad52218ab114f1fec426514e79edf407 at: https://66e6cb30f79a9a88c715ccb7--layer5.netlify.app

**Comment by Ashparshp** [comment]
@THEBOSS0369, Thank you for your contribution!
Let's discuss this during the website call on Monday at 5:30 PM IST (7:00 AM CT).

I'm adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk) || [meeting link](https://meet.layer5.io/websites).

File: src/theme/app/themeStyles.js
Comment (vishalvivekm) [inline src/theme/app/themeStyles.js:1]: We wouldn't want to change theme colors @THEBOSS0369

**Comment by vishalvivekm** [comment]
Also, will you please remove commits not pertinent to the issue ?

**Comment by THEBOSS0369** [comment]
> @THEBOSS0369 
> Sharing the links form our discussion in [websites meeting](https://www.youtube.com/watch?v=7ln7cWOHOKo&t=170s) 
> - https://github.com/layer5io/layer5/blob/b0d781f24c3f323b96a772c61b8446a291896cb2/src/sections/Company/News-grid/NewsGrid.style.js#L102
> - https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#things-to-keep-in-mind-1

Thanks for sharing this!! 🙂

**Comment by THEBOSS0369** [comment]
> @THEBOSS0369 
> Sharing the links form our discussion in [websites meeting](https://www.youtube.com/watch?v=7ln7cWOHOKo&t=170s) 
> - https://github.com/layer5io/layer5/blob/b0d781f24c3f323b96a772c61b8446a291896cb2/src/sections/Company/News-grid/NewsGrid.style.js#L102
> - https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#things-to-keep-in-mind-1

Thanks for sharing this!! 🙂🙂

**Comment by THEBOSS0369** [comment]
> Also, will you please remove commits not pertinent to the issue ?

Sure! I will remove the changes from searchbox. Js and apply the changes in gridestyle. Js

**Comment by vishalvivekm** [comment]
> > Also, will you please remove commits not pertinent to the issue ?
> 
> Sure! I will remove the changes from searchbox. Js and apply the changes in gridestyle. Js

I was referring to these Commits: 
Screenshot_20240917-084243

File: src/reusecore/Search/searchbox.style.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ export const SearchWrapper = styled.div`
       border-radius: 5px;
       border: 1px solid ${(props) => props.theme.headingColor};
       background-color: ${(props) => props.theme.shadowLightColor};
+      
+      /*Changes color for the text*/
+      color: ${(props) => props.theme.textColor};
```
Comment (leecalcote) [inline src/reusecore/Search/searchbox.style.js:20]: Will this affect the search field on /blog, too?

Hopefully, it does as that field is in need of similar dark mode treatment.

Comment (leecalcote) [reply src/reusecore/Search/searchbox.style.js:20]: I see now that it does.

**Comment by THEBOSS0369** [comment]
> > > Also, will you please remove commits not pertinent to the issue ?
> > 
> > Sure! I will remove the changes from searchbox. Js and apply the changes in gridestyle. Js
> 
> I was referring to these Commits: 
> Screenshot_20240917-084243
> 

Ohhhh, i get it now! I didn't commited those changes but can you tell me how do i remove those commits like do i have to revert back or any other way.

**Comment by THEBOSS0369** [comment]
@vishalvivekm Can you confirm that i have to do this right?? 
image

**Comment by THEBOSS0369** [comment]
Hey @vishalvivekm ! Please review the changes i made. Also, now the style is similar as the style of searchbox in /news page.

**Comment by l5io** [comment]
🚀 Preview for commit b263459fbbe55c1764dbfddf18c496752daebcee at: https://66ed04ee857b09e66702c1f3--layer5.netlify.app

**Comment by vishalvivekm** [comment]
@THEBOSS0369 why are we still modifying the themes hex codes?

**Comment by vishalvivekm** [comment]
> @THEBOSS0369 why are we still modifying the themes hex codes?

Please review the feedback: https://www.youtube.com/watch?v=7ln7cWOHOKo&t=170s

**Comment by THEBOSS0369** [comment]
Hey @vishalvivekm ! Apologies for late reply. 
I got confused and wasn't able to figure out how. Now I have sorted everything, I have added the style for Searchbox in Required Page's   ...grid.style.js file and removed the changes from the Themestyles.js and searchbox.style.js . 

Note the Changes you will see in Searchhox.style.js are the changes which I copied from the official searchbox.style.js to revert back to the original color. 

Thanks

**Comment by l5io** [comment]
🚀 Preview for commit df81b5ec70b776de3fdd1eed81111f7332496df7 at: https://66f5bc40a708be4127521040--layer5.netlify.app

**Comment by sudhanshutech** [comment]
@THEBOSS0369 any updates?

**Comment by THEBOSS0369** [comment]
> @THEBOSS0369 any updates?

Hey @sudhanshutech ! This PR is completed and ready to merge. I had exams which ended today so i wasn't active on the slack. 
Thanks

**Comment by sudhanshutech** [comment]
@THEBOSS0369 there was gatsby upgrade in site please ensure you changes again that it works.

**Comment by l5io** [comment]
🚀 Preview for commit defd7355469a972c7a5ecd23f78633d6850df5d7 at: https://6706648dbcd8ce273bf87ec6--layer5.netlify.app

**Comment by THEBOSS0369** [comment]
@sudhanshutech ! Everything is working fine i just checked it here are ss. You can also verify these changes on the preview link.

image
image
image

**Comment by l5io** [comment]
🚀 Preview for commit df8b9487c0aa0c2259754e8f89404a76a1bfe03c at: https://67066a3e750f902dccf909cf--layer5.netlify.app

**Review by sudhanshutech — CHANGES_REQUESTED** [review]
{ACFBDB30-87EE-48D7-9355-82A42F85CF69}
this search icon should also be a white shade as well

**Comment by sudhanshutech** [comment]
> > {ACFBDB30-87EE-48D7-9355-82A42F85CF69}
> > this search icon should also be a white shade as well
> 
> @sudhanshutech ! The style is same as in Layer5/news section and @vishalvivekm told me to copy the same style. Although I agree that it should be white. If you both agree to that I will change it to white.

yes please , for light mode it could stay the same but for dark mode a white shade. See if there other icons which do the same like dark shade for light mode and white shade for light mode and do the same

**Comment by THEBOSS0369** [comment]
@sudhanshutech @vishalvivekm ! I have removed the code from the required searchbox's respective files and add it to the global styles so that in future we don't have to copy paste the whole code.
 **Note**: I have also removed it from the News Section as well . 
Here are the Results!

Screenshot 2024-10-17 121657
Screenshot 2024-10-17 121755
Screenshot 2024-10-17 121831
Screenshot 2024-10-17 121912
Screenshot 2024-10-17 122034

**Comment by l5io** [comment]
🚀 Preview for commit 26dd7b6a429bf8bf7cf94cf552986fd246027c88 at: https://67134b3657a13b407c38a272--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 68842882b272557c634b23a7f729eb58ec2b311b at: https://671632d73284890bdad28f2c--layer5.netlify.app

**Comment by THEBOSS0369** [comment]
Hey @sudhanshutech @vishalvivekm ! I have done the changes you asked. Please look into it and i hope this is the last thing i needed to change. Thanks

image
image

**Comment by l5io** [comment]
🚀 Preview for commit 433f23a42343cf140f1a0d58be12ee82ba6b0810 at: https://67179bdbb7a10f10183228ff--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 59368e32dfc09284a1faa0a3c84a72d33b6467da at: https://671ccd0237fc879c50759566--layer5.netlify.app

**Comment by THEBOSS0369** [comment]
> @THEBOSS0369 build is failing

@sudhanshutech This issue is not from my side. You can clearly see the build previews link before yesterday. I also checked other PR's tests the build test is failing in those PRs as well.

**Comment by THEBOSS0369** [comment]
@vishalvivekm I have moved the code for integration search bar to its own file as you asked me. Please let me know if any changes are needed! Here are the results 

image
image

**Comment by vishalvivekm** [comment]
> > @THEBOSS0369 build is failing
> 
> @sudhanshutech This issue is not from my side. You can clearly see the build previews link before yesterday. I also checked other PR's tests the build test is failing in those PRs as well.

should be fixed now

**Comment by l5io** [comment]
🚀 Preview for commit ae1c6698fca3aa8f70a7b659a1d72724ff8cba23 at: https://6722282a049c71acb7dc5135--layer5.netlify.app

**Review by vishalvivekm — APPROVED** [review]

**Comment by vishalvivekm** [comment]
Thanks much @THEBOSS0369 for your contributions.

**Comment by THEBOSS0369** [comment]
@vishalvivekm ! Thanks for merging this, now i can work on more issues 😁

**Pull request merged**